### PR TITLE
add spring to core module

### PR DESCRIPTION
**Pull request by MarkPikanovsky**
I have excluded config and test class from jacoco, and added sout in TestCoreApp class to avoid errors from pmd plugin...maybe it`s wrong way ... but lets see

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,6 +44,20 @@
...
     <dependencies>
+      <!-- https://mvnrepository.com/artifact/org.springframework/spring-context -->
+      <dependency>
+        <groupId>org.springframework</groupId>
+        <artifactId>spring-context</artifactId>
+        <version>5.1.3.RELEASE</version>
+      </dependency>
+
+      <!-- https://mvnrepository.com/artifact/org.springframework/spring-webmvc -->
+      <dependency>
```
Comment (aleksashyn) [inline pom.xml]: Redundant dependency

File: core/pom.xml [hunk 2]
```diff
@@ -30,6 +30,12 @@
       <artifactId>logback-classic</artifactId>
     </dependency>
 
+    <!-- https://mvnrepository.com/artifact/org.springframework/spring-context -->
```
Comment (aleksashyn) [inline core/pom.xml]: Remove comment

File: core/src/main/java/com/lms/system/teacher/crm/core/CoreRootConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ComponentScan
+public class CoreRootConfiguration {
+
+    @Bean
```
Comment (aleksashyn) [inline core/src/main/java/com/lms/system/teacher/crm/core/CoreRootConfiguration.java]: We decided to used annotation configuration for spring. Please, use Component

File: core/src/main/java/com/lms/system/teacher/crm/core/CoreRootConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.lms.system.teacher.crm.core;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ComponentScan
```
Comment (aleksashyn) [inline core/src/main/java/com/lms/system/teacher/crm/core/CoreRootConfiguration.java]: Please specify package for components scan

File: pom.xml [hunk 5]
```diff
@@ -44,6 +44,20 @@
 
   <dependencyManagement>
     <dependencies>
+      <!-- https://mvnrepository.com/artifact/org.springframework/spring-context -->
```
Comment (aleksashyn) [inline pom.xml]: Please remove comment

File: core/src/main/java/com/lms/system/teacher/crm/core/TestCoreApp.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.lms.system.teacher.crm.core;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+public class TestCoreApp {
```
Comment (aleksashyn) [inline core/src/main/java/com/lms/system/teacher/crm/core/TestCoreApp.java]: Redundant class

File: core/src/main/java/com/lms/system/teacher/crm/core/CoreRootConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+public class CoreRootConfiguration {
```
Comment (aleksashyn) [inline core/src/main/java/com/lms/system/teacher/crm/core/CoreRootConfiguration.java]: Why Root? Do you plan to add one more configuration?

File: core/src/main/java/com/lms/system/teacher/crm/core/CoreConfiguration.java [hunk 8]
```diff
@@ -1,15 +1,10 @@
 package com.lms.system.teacher.crm.core;
```
Comment (aleksashyn) [inline core/src/main/java/com/lms/system/teacher/crm/core/CoreConfiguration.java]: Please add package config for this class

**Comment by MarkPikanovsky** [comment]
> What about tests?
Have no ideas what to test...

**Comment by aleksashyn** [comment]
> > What about tests?
> > Have no ideas what to test...

You can check that Demo bean is created and you can autowired it

File: pom.xml [hunk 9]
```diff
@@ -44,6 +44,12 @@
 
   <dependencyManagement>
     <dependencies>
+      <dependency>
+        <groupId>org.springframework</groupId>
+        <artifactId>spring-context</artifactId>
+        <version>5.1.3.RELEASE</version>
```
Comment (aleksashyn) [inline pom.xml]: please extract version to property

File: pom.xml [hunk 10]
```diff
@@ -226,6 +232,9 @@
                     </limits>
                   </rule>
                 </rules>
+                <excludes>
+                  <exclude>**/*CoreConfiguration.*</exclude>
```
Comment (aleksashyn) [inline pom.xml]: please replace with package `config`

File: core/src/test/java/com/lms/system/teacher/crm/core/DemoTest.java [hunk 11]
```diff
@@ -26,4 +37,13 @@ void should_test_demo() {
         // then
         assertThatThrownBy(Demo::demo).isInstanceOf(IllegalArgumentException.class);
     }
+
+    @Test
+    void should_autowired_demo() {
+        // given
+        // when
+        // then
+        assertThat(demo != null).isTrue();
```
Comment (aleksashyn) [inline core/src/test/java/com/lms/system/teacher/crm/core/DemoTest.java]: `assertThat().notNull()`

**Pull request merged**